mousestyles/dynamics/__init__.py

```diff
@@ -431,3 +431,98 @@ def get_score(true_day, simulated_day, weight=[1, 10, 50, 1]):
             score += weight[int(status)]
     score = score/len_true
     return score
+
+
+def find_best_interval(df, strain_num, interval_length_initial=np.arange(600,
+                       7800, 600)):
+    r"""
+    Returns the optimized time interval length and the corresponding
+    fake mouse behavior string with the evaluation score for a
+    particular mouse strain. The input data are the pandas DataFrame
+    generated by function create_time_matrix which contains all strains
+    information, the strain number that we want to develope, and some
+    initial time interval length we want to optimize on. The outputs
+    are the optimized time interval length, the similated mouse states
+    string using that optimized time interval and the evaluation score
+    comparing this simulated mouse with the real mice behavior in the
+    strain using the same optimized time interval length.
+
+    Parameters
+    ----------
+    df: Pandas.DataFrame
+        a huge data frame containing info on strain, mouse
+        no., mouse day, and different states at chosen time
+        points.
+    starin_num: int
+        an integer specifying the desired mouse strain
+    interval_length_initial: numpy.ndarray
+        a numpy.ndarray specifying the range of time interval
+        that it optimizes on
+
+    Returns
+    -------
+    best_interval_length: int
+        a interger indicating the optimized time interval
+        in terms of the evaluation score
+    best_fake_mouse: list
+        a list of integers indicating the simulated
+        states using the best_interval_length
+    best_score: float
+        a float representing the evaluation score
+        comparing the best_fake_mouse with the real mice
+        behavior in the strain under the same optimized
+        time interval length.
+
+    Examples
+    --------
+    >>> row_i = np.hstack((np.zeros(40)))
+    >>> ime_df_eg = np.vstack((row_i, row_i, row_i))
+    >>> time_df_eg = pd.DataFrame(time_df_eg)
+    >>> time_df_eg.rename(columns={0:'strain'}, inplace=True)
+    >>> find_best_interval(time_df_eg, 0, np.arange(10, 40, 10))
+    >>> (10, array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    0, 0]), 0.98245614035087725)
+"""
+    # check all the inputs
+    condition_df = (type(df) == pd.core.frame.DataFrame)
+    condition_strain_num = (strain_num in (0, 1, 2))
+    condition_interval_length_initial = (type(interval_length_initial) ==
+                                         np.ndarray and
+                                         np.sum(interval_length_initial > 0) ==
+                                         len(interval_length_initial) and
+                                         all(isinstance(i, np.int64) for i
+                                             in interval_length_initial))
+
```

Comment:
may be we can add separative comments to indicate which check the code is running for.